dpsim-villas/src/InterfaceVillasQueueless.cpp

```diff
@@ -0,0 +1,374 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "villas/node/memory.hpp"
+#include "villas/node/memory_type.hpp"
+#include "villas/path.hpp"
+#include "villas/signal_type.hpp"
+#include <dpsim-villas/InterfaceVillasQueueless.h>
+#include <dpsim-villas/InterfaceWorkerVillas.h>
+#include <memory>
+#include <spdlog/spdlog.h>
+#include <typeinfo>
+#include <villas/node.h>
+
+using namespace CPS;
+using namespace villas;
+
+namespace DPsim {
+
+InterfaceVillasQueueless::InterfaceVillasQueueless(
+    const String &nodeConfig, const String &name,
+    spdlog::level::level_enum logLevel)
+    : Interface(name, logLevel), mNodeConfig(nodeConfig), mNode(nullptr),
+      mSamplePool(), mSequenceToDpsim(0), mSequenceFromDpsim(0) {}
+
+void InterfaceVillasQueueless::createNode() {
+  if (villas::node::memory::init(100) != 0) {
+    SPDLOG_LOGGER_ERROR(mLog, "Error: Failed to initialize memory subsystem!");
+    std::exit(1);
+  }
+  json_error_t error;
+  json_t *config = json_loads(mNodeConfig.c_str(), 0, &error);
+  if (config == nullptr) {
+    SPDLOG_LOGGER_ERROR(mLog, "Error: Failed to parse node config! Error: {}",
+                        error.text);
+    throw JsonError(config, error);
+  }
+
+  const json_t *nodeType = json_object_get(config, "type");
+  if (nodeType == nullptr) {
+    SPDLOG_LOGGER_ERROR(mLog, "Error: Node config does not contain type-key!");
+    std::exit(1);
+  }
+  String nodeTypeString = json_string_value(nodeType);
+
+  mNode = node::NodeFactory::make(nodeTypeString);
+
+  int ret = 0;
+  ret = mNode->parse(config);
+  if (ret < 0) {
+    SPDLOG_LOGGER_ERROR(mLog,
+                        "Error: Node in InterfaceVillas failed to parse "
+                        "config. Parse returned code {}",
+                        ret);
+    std::exit(1);
+  }
+  ret = mNode->check();
+  if (ret < 0) {
+    SPDLOG_LOGGER_ERROR(
+        mLog,
+        "Error: Node in InterfaceVillas failed check. Check returned code {}",
+        ret);
+    std::exit(1);
+  }
+  struct villas::node::memory::Type *pool_mt = &villas::node::memory::heap;
```

Comment:
Please try do add some more empty lines. The code looks quite dense is hard to read in my opinion. Especially after closing brackets, I think an empty line would do better.

Reply:
You coding style is also not consistent in this regard. Sometimes you have empty lines after a `}`, and sometimes not.

Reply:
Hm I use clang-format and editorconf. I think if its not specified in clang-format then it should be fine, no?